### PR TITLE
DataSet: trying to import without any columns logs an unhelpful error

### PR DESCRIPTION
**Pull request by nadzpogi**
relates to xibosignage/xibo#3458

### CODE REVIEW

**Review by PeterMis — CHANGES_REQUESTED** [review]
I think this looks fine for manual csv import, however looking at the issue it should also cover remote datasets.

ie processing json/csv from remote dataSet source should also check for no columns - in this case perhaps also conditionally for no `remote` columns.

The message on the form could also be adjusted for remote dataSet to check the remote columns - the message for no columns at all that you have is fine, but in case of the remote dataSet we are more interested in remote columns instead.

File: lib/Helper/DataSetUploadHandler.php [hunk 1]
```diff
@@ -62,6 +62,20 @@ protected function handleFormData($file, $index)
                 throw new AccessDeniedException();
             }
 
+            // Get all columns
+            $columns = $dataSet->getColumn();
+
+            // Filter columns where dataSetColumnType is "Value"
+            $filteredColumns = array_filter($columns, function ($column) {
+                return $column->dataSetColumnType == 'Value';
```
Comment (PeterMis) [inline lib/Helper/DataSetUploadHandler.php:70]: Might be better to rely on `dataSetColumnTypeId` instead? ( result should be the same )

File: lib/Helper/DataSetUploadHandler.php [hunk 2]
```diff
@@ -62,6 +62,20 @@ protected function handleFormData($file, $index)
...
+            });
+
+            // Check if there are any value columns defined in the dataset
+            if (count($filteredColumns) === 0) {
+                $controller->getLog()->error('Import failed: No value columns defined in the dataset.');
+                throw new InvalidArgumentException('Import failed: No value columns defined in the dataset.');
```
Comment (PeterMis) [inline lib/Helper/DataSetUploadHandler.php:76]: Should we translate the exception message?

File: lib/XTR/RemoteDataSetFetchTask.php [hunk 3]
```diff
@@ -99,6 +99,20 @@ public function run()
                     continue;
                 }
 
+                // Get all columns
+                $columns = $dataSet->getColumn();
+
+                // Filter columns where dataSetColumnType is "Remote"
+                $filteredColumns = array_filter($columns, function ($column) {
+                    return $column->dataSetColumnType == 'Remote';
```
Comment (PeterMis) [inline lib/XTR/RemoteDataSetFetchTask.php:107]: Similar to previous comment, perhaps rely on `dataSetColumnTypeId` ?

**Review by PeterMis — COMMENTED** [review]
Looks good to me, just a very minor comments below.

**Review by PeterMis — APPROVED** [review]

**Comment by ifarzana** [comment]
TEST: /Administration END

**Pull request merged**